cmd/otelopscol/main.go

```diff
@@ -41,13 +45,45 @@ func main() {
 		Version:     version.Version,
 	}
 
-	params := service.CollectorSettings{Factories: factories, BuildInfo: info}
+	params := service.CollectorSettings{
+		Factories: factories,
+		BuildInfo: info,
+		LoggingOptions: []zap.Option{
+			logSpamFilterCore(),
+		},
+	}
 
 	if err := run(params); err != nil {
 		log.Fatal(err)
 	}
 }
 
+// Returns a zapfilter core that will filter log spam from the otel collector.
+// Upstream issue: https://github.com/open-telemetry/opentelemetry-collector/issues/3004
+func logSpamFilterCore() zap.Option {
+	logFilterFunc := func(entry zapcore.Entry, fields []zapcore.Field) bool {
+		if !strings.Contains(entry.Caller.File, "scrapercontroller.go") {
+			return true
+		}
+		for _, field := range fields {
+			if field.Key == "error" {
+				logError, ok := field.Interface.(error)
+				if !ok {
+					return true
+				}
+				return !strings.Contains(
+					logError.Error(),
+					"error reading process name for pid")
```

Comment:
Can we make this a constant slice (aka a list of strings that store the error messages we want to exclude)? The benefit is that it's easier to add additional filter rules.

e.g.

```
func logMessagesToExclude() []string {
    return []string{
        "error reading process name for pid",
    }
}
```

Reply:
If you do, make it a global variable, not a func.

Reply:
I made this a more generic abstraction that will accept a list of error message substrings and made the comment referencing the issue only on that particular substring.